From ae80234b10972d1050d4e41f30bd1212493de544 Mon Sep 17 00:00:00 2001
From: Xiang Wang <xiang.a.wang@intel.com>
Date: Thu, 4 Feb 2016 14:45:12 +0800
Subject: [PATCH 08/13] REVERTME i2c: designware: decrease abort_source logs

In previous patch (https://android.intel.com/#/c/457298/), logs
will be printed when i2c abort (e.g. data no ACK) for debugging.
Usually, we have no such abort. However, when camera works, we
can see pretty much data no ACK which introduced many error logs.
As a result, camera CTS fails.

So we can only print abort_source when more serious abort happens.

Change-Id: I42e0bb10f0fd4c69b1bce3f8f1bf4a94b99bf596
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18965
Signed-off-by: Xiang Wang <xiang.a.wang@intel.com>
Reviewed-on: https://android.intel.com:443/466344
---
 drivers/i2c/busses/i2c-designware-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-designware-core.c b/drivers/i2c/busses/i2c-designware-core.c
index 1c13a68..4b9fcfa 100644
--- a/drivers/i2c/busses/i2c-designware-core.c
+++ b/drivers/i2c/busses/i2c-designware-core.c
@@ -641,7 +641,6 @@ static int i2c_dw_handle_tx_abort(struct dw_i2c_dev *dev)
 	unsigned long abort_source = dev->abort_source;
 	int i;
 
-	dev_err(dev->dev, "abort_source = 0x%08lx\n", abort_source);
 	if (abort_source & DW_IC_TX_ABRT_NOACK) {
 		for_each_set_bit(i, &abort_source, ARRAY_SIZE(abort_sources))
 			dev_dbg(dev->dev,
@@ -649,6 +648,7 @@ static int i2c_dw_handle_tx_abort(struct dw_i2c_dev *dev)
 		return -EREMOTEIO;
 	}
 
+	dev_err(dev->dev, "abort_source = 0x%08lx\n", abort_source);
 	for_each_set_bit(i, &abort_source, ARRAY_SIZE(abort_sources))
 		dev_err(dev->dev, "%s: %s\n", __func__, abort_sources[i]);
 
-- 
1.9.1

